hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/RootPageDisplayFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.ozone.s3;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * This redirect helps to show and info page in case the endpoint is opened
+ * from the browser.
+ */
+public class RootPageDisplayFilter implements Filter {
+
+  @Override
+  public void init(FilterConfig filterConfig) throws ServletException {
+
+  }
+
+  @Override
+  public void doFilter(ServletRequest servletRequest,
+      ServletResponse servletResponse, FilterChain filterChain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+    String httpMethod = httpRequest.getMethod();
+    String uri = httpRequest.getRequestURI();
+    String authorizationHeader = httpRequest.getHeader("Authorization");
+    if (httpMethod.equalsIgnoreCase("GET") && authorizationHeader == null && uri
+        .equals("/")) {
+      ((HttpServletResponse) servletResponse).sendRedirect("/static/");
+    } else {
```

Comment:
What does else part do, for other HTTP method requests?
Or we should fail the request as it is not a valid request.
Or I am not sure if my understanding is correct, could you provide some info here.

Reply:
`filterChain.doFilter` (else branch) means: "do all the other stuff defined by other filters/servlets/jaxrs classes.

This condition has a short-circuit response for the "GET /" page (without Authorization header). In this case it immediately redirect to the internal documentation page.

For any other case it does the normal work. (else)